spinalcordtoolbox/scripts/sct_run_batch.py

```diff
@@ -174,6 +175,33 @@ def _find_nonsys32_bash_exe():
     return shutil.which('bash', path=nonsys32_paths)
 
 
+def _filter_directories(dir_list, include=None, include_list=None, exclude=None, exclude_list=None):
+    # Handle inclusion lists
+    if include is not None:
+        dir_list = [f for f in dir_list if re.search(include, f) is not None]
+
+    if include_list is not None:
+        # TODO decide if we should warn users if one of their inclusions isn't around
+        dir_list = [f for f in dir_list
+                    # Check if include_list specified entire path (e.g. "sub-01/ses-01")
+                    if f in include_list
+                    # Check if include_list specified a subdirectory (e.g. just "sub-01" or just "ses-01")
+                    or any([p in include_list for p in pathlib.Path(f).parts])]
```

Comment:
I like this list comprehension! And the use of `pathlib.Path(f).parts`, very clear. Nitpick: you can get rid of the inner list comprehension, and just write `any(p in include_list for ...)`.